moveit_core/kinematic_constraints/src/utils.cpp

```diff
@@ -281,263 +283,248 @@ moveit_msgs::msg::Constraints constructGoalConstraints(const std::string& link_n
-//     else if (it->first == "target_pose")
-//     {
-//       if (!constructPoseStamped(it, constraint.target_pose))
-//         return false;
-//     }
-//     else if (it->first == "cone_sides")
-//       constraint.cone_sides = static_cast<int>(it->second);
-//     else if (it->first == "sensor_pose")
-//     {
-//       if (!constructPoseStamped(it, constraint.sensor_pose))
-//         return false;
-//     }
-//     else if (it->first == "max_view_angle")
-//       constraint.max_view_angle = parseDouble(it->second);
-//     else if (it->first == "max_range_angle")
-//       constraint.max_range_angle = parseDouble(it->second);
-//     else
-//     {
-//       ROS_WARN_STREAM_NAMED(LOGNAME, "orientation constraint contains unknown entity '" << it->first << "'");
-//     }
-//   }
-//
-//   constraint.sensor_view_direction = moveit_msgs::VisibilityConstraint::SENSOR_X;
-//
-//   return true;
-// }
-//
-// static bool collectConstraints(XmlRpc::XmlRpcValue& params, moveit_msgs::Constraints& constraints)
-// {
-//   if (params.getType() != XmlRpc::XmlRpcValue::TypeArray)
-//   {
-//     ROS_ERROR_NAMED(LOGNAME, "expected constraints as array");
-//     return false;
-//   }
-//
-//   for (int i = 0; i < params.size(); ++i)  // NOLINT(modernize-loop-convert)
-//   {
-//     if (!params[i].hasMember("type"))
-//     {
-//       ROS_ERROR_NAMED(LOGNAME, "constraint parameter does not specify its type");
-//     }
-//     else if (params[i]["type"] == "joint")
-//     {
-//       constraints.joint_constraints.emplace_back();
-//       if (!constructConstraint(params[i], constraints.joint_constraints.back()))
-//         return false;
-//     }
-//     else if (params[i]["type"] == "position")
-//     {
-//       constraints.position_constraints.emplace_back();
-//       if (!constructConstraint(params[i], constraints.position_constraints.back()))
-//         return false;
-//     }
-//     else if (params[i]["type"] == "orientation")
-//     {
-//       constraints.orientation_constraints.emplace_back();
-//       if (!constructConstraint(params[i], constraints.orientation_constraints.back()))
-//         return false;
-//     }
-//     else if (params[i]["type"] == "visibility")
-//     {
-//       constraints.visibility_constraints.emplace_back();
-//       if (!constructConstraint(params[i], constraints.visibility_constraints.back()))
-//         return false;
-//     }
-//   }
-//
-//   return true;
-// }
-//
-// bool constructConstraints(XmlRpc::XmlRpcValue& params, moveit_msgs::Constraints& constraints)
-// {
-//   if (!isStruct(params, { "name", "constraints" }, "Parameter"))
-//     return false;
-//
-//   constraints.name = static_cast<std::string>(params["name"]);
-//   return collectConstraints(params["constraints"], constraints);
-// }
-// }  // namespace kinematic_constraints
+  std::vector<double> orientation;
+  if (!node->get_parameter(pose_param + "orientation", orientation) || orientation.size() != 3)
+    return false;
```

Comment:
There is logging that warns about wrong format. As this is a straight port I don't think it makes sense to add too much extra logging, especially since we should switch to declaring parameters which would allow better and more verbose checking at runtime.